tests/PHPStan/Analyser/data/filter_var.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace FilterVar;
+
+use function PHPStan\Testing\assertType;
+
+function doBar($mixed) {
+	if (filter_var($mixed, FILTER_VALIDATE_FLOAT)) {
+		assertType('float', $mixed);
+	}
+	assertType('mixed', $mixed);
+
+	if (filter_var($mixed, FILTER_VALIDATE_FLOAT) === true) {
+		assertType('float', $mixed);
+	}
+	assertType('mixed', $mixed);
+
+	if (filter_var($mixed, FILTER_VALIDATE_FLOAT) === false) {
+		assertType('mixed', $mixed); // could be mixed~float
```

Comment:
did not implement as I think its a very exotic use-case for filter_var

Reply:
I don't think you can just do `!$context->null()` here.

If `filter_var($mixed, FILTER_VALIDATE_FLOAT)` is false then you can say it cannot be a `numeric-string` for example.

I think this is wrong too:

```
	if (filter_var($mixed, FILTER_VALIDATE_FLOAT) === true) {
		assertType('float', $mixed);
	}
```

You're treating this function as `is_float()` but it's not that. It RETURNS a float (https://3v4l.org/lQlRY), but from a type-narrowing PoV, the thing remains its original type, you can just say it might be a float, or it might be a numeric string for example: https://3v4l.org/2b3HN

Because it returns a float, the test `filter_var($mixed, FILTER_VALIDATE_FLOAT) === true` is actually wrong too...